File: Backend/BooksAPI/Models/Book.cs
using BooksAPI.Data.Entities;
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations.Schema;

namespace BooksAPI.Models
{
    public class Book
    {
        public Guid Id { get; set; }


        [ForeignKeyAttribute("User")]
        public Guid UserId { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public string Publishing { get; set; }

        //       public DateTime PublicationDate { get; set; }
        public int Page { get; set; }

        //      public string ISBN { get; set; }
        public string Language { get; set; }
        public string Status { get; set; }
       public IFormFile Image { get; set; }
        public string Category { get; set; }
    }
}

File: Frontend/src/app/models/book.ts
import { Category } from './category';
import { User } from './user';

export class Book {
  isbn: string = '';
  id: string = '';
  userId: string = '';
  categoryId: string = '';
  title: string = '';
  author: string = '';
  publisher: string = '';
  publicationDate: Date = new Date();
  page: number = 0;
  price: number = 0;
  language: string = '';
  condition: string = '';
  isSold: boolean = false;
  image: string = '';
  category: Category = new Category();
  user: User = new User();

  constructor(init?: Partial<Book>) {
    Object.assign(this, init);
  }
}
